### ISSUE TITLE
An error occurs when running tutorial: T01-MACE-Practice-I

### ISSUE THREAD

**Issue by mast1986** [ACEsuit/mace#600]
**Describe the bug**
When running part 4: Molecular Dynamics with MACE, will be IndexError: too many indices for tensor of dimension 1.

**To Reproduce**
1. Go to 'T01-MACE-Practice-I.ipynb'
2. Execute jupyter cells sequentially, when running part 4, second cell, see error

**Screenshots**
![image](https://github.com/user-attachments/assets/ff0db2dd-3e0a-4e20-9603-0d39ee7b572a)

**Desktop (please complete the following information):**
 - OS: [Ubuntu 20.04.6 LTS]
 - mace-torch [0.3.7]
 - CUDA [12.1]
 - GPU Driver [550.54.15]
 - GPU [v100]
 - torch [2.1.0+cu121]

**Comment by ilyes319** [comment]
Are you executing the notebook locally? What repo did you clone?

**Comment by mast1986** [comment]
> Are you executing the notebook locally? What repo did you clone?

Yeah, I run it locally.

The repo info is as follow:
```
!git clone https://github.com/imagdau/Tutorials.git
!pip install xtb nglview ipywidgets rdkit x3dase
!pip install git+https://github.com/imagdau/aseMolec@main
!pip install git+https://github.com/acesuit/mace@develop
```
```
Collecting git+https://github.com/acesuit/mace@develop
  Cloning https://github.com/acesuit/mace (to revision develop) to /tmp/pip-req-build-zj35awyi
  Running command git clone --filter=blob:none --quiet https://github.com/acesuit/mace /tmp/pip-req-build-zj35awyi
  Running command git checkout -b develop --track origin/develop
  Switched to a new branch 'develop'
  Branch 'develop' set up to track remote branch 'develop' from 'origin'.
  Resolved https://github.com/acesuit/mace to commit 7a475fbed5f7e90bb9ec7612015b7909550d92c6
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
  ```

**Comment by ilyes319** [comment]
Can you install the main branch instead?

**Comment by mast1986** [comment]
Sure, I have tried switching to the main branch, but the error is the same.
Some debug info, I try to print some variables shape:
```
atomic_energies shape: torch.Size([12])
node_attrs shape: torch.Size([12, 3])
batch shape: torch.Size([12])
num_atoms_arange: torch.Size([12])
node_heads: torch.Size([12])
```

**Comment by mstapelberg** [comment]
Can confirm I also have this issue with any model trained on mace!=0.3.7 when using mace=0.3.7

As a temporary fix, switching back to mace=0.3.6 fixes this issue!

**Comment by sitengm** [comment]
> Can confirm I also have this issue with any model trained on mace!=0.3.7 when using mace=0.3.7. The same happened when running models trained with mace=0.3.7 in mace=0.3.6. Is there perhaps a new way in 0.3.7 for handling nodes?
> 
> As a temporary fix, switching back to mace=0.3.6 has fixed this issue for me

Thank you, switching back to mace=0.3.6 also work for me.

**Comment by mast1986** [comment]
> Can confirm I also have this issue with any model trained on mace!=0.3.7 when using mace=0.3.7. The same happened when running models trained with mace=0.3.7 in mace=0.3.6. Is there perhaps a new way in 0.3.7 for handling nodes?
> 
> As a temporary fix, switching back to mace=0.3.6 has fixed this issue for me

Thank you, switching back to mace=0.3.6 also work for me.

**Comment by ilyes319** [comment]
Can you send me you 0.3.6 model so I can check how to fix that.

Ilyes Batatia
________________________________

**Comment by mstapelberg** [comment]
Hi Ilyes, 

I have attached a model trained in mace-0.3.7 and a model trained in mace-0.3.6 for reference (labeled as such) in a tar.gz.

When running a 0.3.6 trained model in mace-0.3.7 I get : 
IndexError: too many indices for tensor of dimension 1

Full Error Log:
[mace_0-3-6_on_0-3-7.txt](https://github.com/user-attachments/files/17180185/mace_0-3-6_on_0-3-7.txt)



When running a 0.3.7 model in mace-0.3.6 I get:
RuntimeError: expand(torch.DoubleTensor{[63, 63]}, size=[63]): the number of sizes provided (1) must be greater or equal to the number of dimensions in the tensor (2)

Full Error Log: 
[mace_0-3-7_on_0-3-6.txt](https://github.com/user-attachments/files/17180180/mace_0-3-7_on_0-3-6.txt)


[mace_models_036_037.tar.gz](https://github.com/user-attachments/files/17180173/mace_models_036_037.tar.gz)

Thanks,
Myles

**Comment by ilyes319** [comment]
@mast1986 Sorry for the delay. I think the problem is that you are using the model called _compiled that is not backward compatbile. You should try to use the model that is named `MACE_models/mace01_run-123_stagetwo.model`. Can you confirm that models trained with 0.3.7 are running with 0.3.7?

**Comment by mstapelberg** [comment]
Hi @ilyes319 

I tested a non-compiled model that was trained with Mace-0.3.6 in Mace-0.3.7

I get the following error (when running it on a cpu):

``` bash
{
	"name": "NotImplementedError",
	"message": "Could not run 'aten::empty_strided' with arguments from the 'CUDA' backend. This could be because the operator doesn't exist for this backend, or was omitted during the selective/custom build process (if using custom build). If you are a Facebook employee using PyTorch on mobile, please visit https://fburl.com/ptmfixes for possible resolutions. 'aten::empty_strided' is only available for these backends: [CPU, MPS, Meta, QuantizedCPU, BackendSelect, Python, FuncTorchDynamicLayerBackMode, Functionalize, Named, Conjugate, Negative, ZeroTensor, ADInplaceOrView, AutogradOther, AutogradCPU, AutogradCUDA, AutogradHIP, AutogradXLA, AutogradMPS, AutogradIPU, AutogradXPU, AutogradHPU, AutogradVE, AutogradLazy, AutogradMTIA, AutogradPrivateUse1, AutogradPrivateUse2, AutogradPrivateUse3, AutogradMeta, AutogradNestedTensor, Tracer, AutocastCPU, AutocastXPU, AutocastCUDA, FuncTorchBatched, BatchedNestedTensor, FuncTorchVmapMode, Batched, VmapMode, FuncTorchGradWrapper, PythonTLSSnapshot, FuncTorchDynamicLayerFrontMode, PreDispatch, PythonDispatcher].

CPU: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/build/aten/src/ATen/RegisterCPU.cpp:30455 [kernel]
MPS: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/build/aten/src/ATen/RegisterMPS.cpp:27702 [kernel]
Meta: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/build/aten/src/ATen/RegisterMeta.cpp:26993 [kernel]
QuantizedCPU: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/build/aten/src/ATen/RegisterQuantizedCPU.cpp:951 [kernel]
BackendSelect: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/build/aten/src/ATen/RegisterBackendSelect.cpp:815 [kernel]
Python: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/aten/src/ATen/core/PythonFallbackKernel.cpp:153 [backend fallback]
FuncTorchDynamicLayerBackMode: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/aten/src/ATen/functorch/DynamicLayer.cpp:497 [backend fallback]
Functionalize: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/aten/src/ATen/FunctionalizeFallbackKernel.cpp:349 [backend fallback]
Named: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/aten/src/ATen/core/NamedRegistrations.cpp:7 [backend fallback]
Conjugate: fallthrough registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/aten/src/ATen/ConjugateFallback.cpp:21 [kernel]
Negative: fallthrough registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/aten/src/ATen/native/NegateFallback.cpp:22 [kernel]
ZeroTensor: fallthrough registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/aten/src/ATen/ZeroTensorFallback.cpp:90 [kernel]
ADInplaceOrView: fallthrough registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/aten/src/ATen/core/VariableFallbackKernel.cpp:86 [backend fallback]
AutogradOther: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/torch/csrc/autograd/generated/VariableType_2.cpp:19857 [autograd kernel]
AutogradCPU: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/torch/csrc/autograd/generated/VariableType_2.cpp:19857 [autograd kernel]
AutogradCUDA: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/torch/csrc/autograd/generated/VariableType_2.cpp:19857 [autograd kernel]
AutogradHIP: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/torch/csrc/autograd/generated/VariableType_2.cpp:19857 [autograd kernel]
AutogradXLA: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/torch/csrc/autograd/generated/VariableType_2.cpp:19857 [autograd kernel]
AutogradMPS: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/torch/csrc/autograd/generated/VariableType_2.cpp:19857 [autograd kernel]
AutogradIPU: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/torch/csrc/autograd/generated/VariableType_2.cpp:19857 [autograd kernel]
AutogradXPU: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/torch/csrc/autograd/generated/VariableType_2.cpp:19857 [autograd kernel]
AutogradHPU: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/torch/csrc/autograd/generated/VariableType_2.cpp:19857 [autograd kernel]
AutogradVE: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/torch/csrc/autograd/generated/VariableType_2.cpp:19857 [autograd kernel]
AutogradLazy: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/torch/csrc/autograd/generated/VariableType_2.cpp:19857 [autograd kernel]
AutogradMTIA: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/torch/csrc/autograd/generated/VariableType_2.cpp:19857 [autograd kernel]
AutogradPrivateUse1: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/torch/csrc/autograd/generated/VariableType_2.cpp:19857 [autograd kernel]
AutogradPrivateUse2: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/torch/csrc/autograd/generated/VariableType_2.cpp:19857 [autograd kernel]
AutogradPrivateUse3: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/torch/csrc/autograd/generated/VariableType_2.cpp:19857 [autograd kernel]
AutogradMeta: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/torch/csrc/autograd/generated/VariableType_2.cpp:19857 [autograd kernel]
AutogradNestedTensor: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/torch/csrc/autograd/generated/VariableType_2.cpp:19857 [autograd kernel]
Tracer: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/torch/csrc/autograd/generated/TraceType_2.cpp:17623 [kernel]
AutocastCPU: fallthrough registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/aten/src/ATen/autocast_mode.cpp:209 [backend fallback]
AutocastXPU: fallthrough registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/aten/src/ATen/autocast_mode.cpp:351 [backend fallback]
AutocastCUDA: fallthrough registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/aten/src/ATen/autocast_mode.cpp:165 [backend fallback]
FuncTorchBatched: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/aten/src/ATen/functorch/LegacyBatchingRegistrations.cpp:731 [backend fallback]
BatchedNestedTensor: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/aten/src/ATen/functorch/LegacyBatchingRegistrations.cpp:758 [backend fallback]
FuncTorchVmapMode: fallthrough registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/aten/src/ATen/functorch/VmapModeRegistrations.cpp:27 [backend fallback]
Batched: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/aten/src/ATen/LegacyBatchingRegistrations.cpp:1075 [backend fallback]
VmapMode: fallthrough registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/aten/src/ATen/VmapModeRegistrations.cpp:33 [backend fallback]
FuncTorchGradWrapper: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/aten/src/ATen/functorch/TensorWrapper.cpp:207 [backend fallback]
PythonTLSSnapshot: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/aten/src/ATen/core/PythonFallbackKernel.cpp:161 [backend fallback]
FuncTorchDynamicLayerFrontMode: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/aten/src/ATen/functorch/DynamicLayer.cpp:493 [backend fallback]
PreDispatch: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/aten/src/ATen/core/PythonFallbackKernel.cpp:165 [backend fallback]
PythonDispatcher: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/aten/src/ATen/core/PythonFallbackKernel.cpp:157 [backend fallback]
",
	"stack": "---------------------------------------------------------------------------
NotImplementedError                       Traceback (most recent call last)
Cell In[3], line 46
      1 #calculator = NequIPCalculator.from_deployed_model(
      2         #model_path=potential_path,
      3         #species_to_type_name=species,
   (...)
     40 #potential_path = '../Potentials/vcrti_neb_100_111_nos_vac_33per_e1_f10_efs_complex_compiled_300_500.model'
     41 #potential_path = '../Potentials/vcrti_neb_e1_f10_s25_100_111_nooutsim_efs_engrad_stagetwo_compiled.model'
     43 potential_path = '../Potentials/vcrti_neb_e1_f50_s25_ase_samp_25per_lr005_complex_stagetwo.model'
---> 46 calculator = MACECalculator(model_paths=[potential_path], device='cpu', default_dtype='float32')

File /opt/homebrew/Caskroom/miniforge/base/envs/mace/lib/python3.11/site-packages/mace/calculators/mace.py:127, in MACECalculator.__init__(self, model_paths, device, energy_units_to_eV, length_units_to_A, default_dtype, charges_key, model_type, compile_mode, fullgraph, **kwargs)
    125     self.use_compile = True
    126 else:
--> 127     self.models = [
    128         torch.load(f=model_path, map_location=device)
    129         for model_path in model_paths
    130     ]
    131     self.use_compile = False
    132 for model in self.models:

File /opt/homebrew/Caskroom/miniforge/base/envs/mace/lib/python3.11/site-packages/mace/calculators/mace.py:128, in <listcomp>(.0)
    125     self.use_compile = True
    126 else:
    127     self.models = [
--> 128         torch.load(f=model_path, map_location=device)
    129         for model_path in model_paths
    130     ]
    131     self.use_compile = False
    132 for model in self.models:

File /opt/homebrew/Caskroom/miniforge/base/envs/mace/lib/python3.11/site-packages/torch/serialization.py:1097, in load(f, map_location, pickle_module, weights_only, mmap, **pickle_load_args)
   1095             except RuntimeError as e:
   1096                 raise pickle.UnpicklingError(_get_wo_message(str(e))) from None
-> 1097         return _load(
   1098             opened_zipfile,
   1099             map_location,
   1100             pickle_module,
   1101             overall_storage=overall_storage,
   1102             **pickle_load_args,
   1103         )
   1104 if mmap:
   1105     f_name = \"\" if not isinstance(f, str) else f\"{f}, \"

File /opt/homebrew/Caskroom/miniforge/base/envs/mace/lib/python3.11/site-packages/torch/serialization.py:1525, in _load(zip_file, map_location, pickle_module, pickle_file, overall_storage, **pickle_load_args)
   1522 # Needed for tensors where storage device and rebuild tensor device are
   1523 # not connected (wrapper subclasses and tensors rebuilt using numpy)
   1524 torch._utils._thread_local_state.map_location = map_location
-> 1525 result = unpickler.load()
   1526 del torch._utils._thread_local_state.map_location
   1528 torch._utils._validate_loaded_sparse_tensors()

File /opt/homebrew/Caskroom/miniforge/base/envs/mace/lib/python3.11/site-packages/e3nn/util/codegen/_mixin.py:109, in CodeGenMixin.__setstate__(self, d)
    107 # Load the TorchScript IR buffer
    108 buffer = io.BytesIO(buffer)
--> 109 smod = torch.jit.load(buffer)
    110 assert isinstance(smod, torch.jit.ScriptModule)
    111 # Add the ScriptModule as a submodule

File /opt/homebrew/Caskroom/miniforge/base/envs/mace/lib/python3.11/site-packages/torch/jit/_serialization.py:165, in load(f, map_location, _extra_files, _restore_shapes)
    163     cpp_module = torch._C.import_ir_module(cu, os.fspath(f), map_location, _extra_files, _restore_shapes)  # type: ignore[call-arg]
    164 else:
--> 165     cpp_module = torch._C.import_ir_module_from_buffer(
    166         cu, f.read(), map_location, _extra_files, _restore_shapes
    167     )  # type: ignore[call-arg]
    169 # TODO: Pretty sure this approach loses ConstSequential status and such
    170 return wrap_cpp_module(cpp_module)

NotImplementedError: Could not run 'aten::empty_strided' with arguments from the 'CUDA' backend. This could be because the operator doesn't exist for this backend, or was omitted during the selective/custom build process (if using custom build). If you are a Facebook employee using PyTorch on mobile, please visit https://fburl.com/ptmfixes for possible resolutions. 'aten::empty_strided' is only available for these backends: [CPU, MPS, Meta, QuantizedCPU, BackendSelect, Python, FuncTorchDynamicLayerBackMode, Functionalize, Named, Conjugate, Negative, ZeroTensor, ADInplaceOrView, AutogradOther, AutogradCPU, AutogradCUDA, AutogradHIP, AutogradXLA, AutogradMPS, AutogradIPU, AutogradXPU, AutogradHPU, AutogradVE, AutogradLazy, AutogradMTIA, AutogradPrivateUse1, AutogradPrivateUse2, AutogradPrivateUse3, AutogradMeta, AutogradNestedTensor, Tracer, AutocastCPU, AutocastXPU, AutocastCUDA, FuncTorchBatched, BatchedNestedTensor, FuncTorchVmapMode, Batched, VmapMode, FuncTorchGradWrapper, PythonTLSSnapshot, FuncTorchDynamicLayerFrontMode, PreDispatch, PythonDispatcher].

CPU: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/build/aten/src/ATen/RegisterCPU.cpp:30455 [kernel]
MPS: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/build/aten/src/ATen/RegisterMPS.cpp:27702 [kernel]
Meta: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/build/aten/src/ATen/RegisterMeta.cpp:26993 [kernel]
QuantizedCPU: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/build/aten/src/ATen/RegisterQuantizedCPU.cpp:951 [kernel]
BackendSelect: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/build/aten/src/ATen/RegisterBackendSelect.cpp:815 [kernel]
Python: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/aten/src/ATen/core/PythonFallbackKernel.cpp:153 [backend fallback]
FuncTorchDynamicLayerBackMode: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/aten/src/ATen/functorch/DynamicLayer.cpp:497 [backend fallback]
Functionalize: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/aten/src/ATen/FunctionalizeFallbackKernel.cpp:349 [backend fallback]
Named: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/aten/src/ATen/core/NamedRegistrations.cpp:7 [backend fallback]
Conjugate: fallthrough registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/aten/src/ATen/ConjugateFallback.cpp:21 [kernel]
Negative: fallthrough registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/aten/src/ATen/native/NegateFallback.cpp:22 [kernel]
ZeroTensor: fallthrough registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/aten/src/ATen/ZeroTensorFallback.cpp:90 [kernel]
ADInplaceOrView: fallthrough registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/aten/src/ATen/core/VariableFallbackKernel.cpp:86 [backend fallback]
AutogradOther: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/torch/csrc/autograd/generated/VariableType_2.cpp:19857 [autograd kernel]
AutogradCPU: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/torch/csrc/autograd/generated/VariableType_2.cpp:19857 [autograd kernel]
AutogradCUDA: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/torch/csrc/autograd/generated/VariableType_2.cpp:19857 [autograd kernel]
AutogradHIP: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/torch/csrc/autograd/generated/VariableType_2.cpp:19857 [autograd kernel]
AutogradXLA: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/torch/csrc/autograd/generated/VariableType_2.cpp:19857 [autograd kernel]
AutogradMPS: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/torch/csrc/autograd/generated/VariableType_2.cpp:19857 [autograd kernel]
AutogradIPU: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/torch/csrc/autograd/generated/VariableType_2.cpp:19857 [autograd kernel]
AutogradXPU: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/torch/csrc/autograd/generated/VariableType_2.cpp:19857 [autograd kernel]
AutogradHPU: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/torch/csrc/autograd/generated/VariableType_2.cpp:19857 [autograd kernel]
AutogradVE: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/torch/csrc/autograd/generated/VariableType_2.cpp:19857 [autograd kernel]
AutogradLazy: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/torch/csrc/autograd/generated/VariableType_2.cpp:19857 [autograd kernel]
AutogradMTIA: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/torch/csrc/autograd/generated/VariableType_2.cpp:19857 [autograd kernel]
AutogradPrivateUse1: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/torch/csrc/autograd/generated/VariableType_2.cpp:19857 [autograd kernel]
AutogradPrivateUse2: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/torch/csrc/autograd/generated/VariableType_2.cpp:19857 [autograd kernel]
AutogradPrivateUse3: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/torch/csrc/autograd/generated/VariableType_2.cpp:19857 [autograd kernel]
AutogradMeta: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/torch/csrc/autograd/generated/VariableType_2.cpp:19857 [autograd kernel]
AutogradNestedTensor: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/torch/csrc/autograd/generated/VariableType_2.cpp:19857 [autograd kernel]
Tracer: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/torch/csrc/autograd/generated/TraceType_2.cpp:17623 [kernel]
AutocastCPU: fallthrough registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/aten/src/ATen/autocast_mode.cpp:209 [backend fallback]
AutocastXPU: fallthrough registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/aten/src/ATen/autocast_mode.cpp:351 [backend fallback]
AutocastCUDA: fallthrough registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/aten/src/ATen/autocast_mode.cpp:165 [backend fallback]
FuncTorchBatched: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/aten/src/ATen/functorch/LegacyBatchingRegistrations.cpp:731 [backend fallback]
BatchedNestedTensor: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/aten/src/ATen/functorch/LegacyBatchingRegistrations.cpp:758 [backend fallback]
FuncTorchVmapMode: fallthrough registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/aten/src/ATen/functorch/VmapModeRegistrations.cpp:27 [backend fallback]
Batched: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/aten/src/ATen/LegacyBatchingRegistrations.cpp:1075 [backend fallback]
VmapMode: fallthrough registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/aten/src/ATen/VmapModeRegistrations.cpp:33 [backend fallback]
FuncTorchGradWrapper: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/aten/src/ATen/functorch/TensorWrapper.cpp:207 [backend fallback]
PythonTLSSnapshot: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/aten/src/ATen/core/PythonFallbackKernel.cpp:161 [backend fallback]
FuncTorchDynamicLayerFrontMode: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/aten/src/ATen/functorch/DynamicLayer.cpp:493 [backend fallback]
PreDispatch: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/aten/src/ATen/core/PythonFallbackKernel.cpp:165 [backend fallback]
PythonDispatcher: registered at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1724788636145/work/aten/src/ATen/core/PythonFallbackKernel.cpp:157 [backend fallback]
"
}
```

Running a 0.3.7 non-compiled model on 0.3.6 gives the same error with "Could not run 'aten::empty_strided' for the CUDA backend"

Granted I'm deploying a model trained on a CUDA device on a CPU system. Can try this on a CUDA system later this week when I have access to one. 

Best,
Myles

**Comment by mast1986** [comment]
> @mast1986 Sorry for the delay. I think the problem is that you are using the model called _compiled that is not backward compatbile. You should try to use the model that is named `MACE_models/mace01_run-123_stagetwo.model`. Can you confirm that models trained with 0.3.7 are running with 0.3.7?

@ilyes319
I apologize for the delayed response due to the National Day holiday. I followed your suggestion and confirmed that it works.

However, I noticed a change between the current 0.3.7 version and the previous one. Previously, the 0.3.7 version would generate models with the naming convention swa.model during training. Now, with the current 0.3.7 version, the models are being generated with the naming convention stagetwo.model.

Thank you
